src/v/cluster/tests/autocreate_test.cc

```diff
@@ -99,48 +97,38 @@ FIXTURE_TEST(
     }
 
     BOOST_REQUIRE_EQUAL(results.size(), 3);
-    wait_for_metadata(get_local_cache(), results);
+    wait_for_metadata(app->metadata_cache.local(), results);
     for (auto& r : results) {
         BOOST_REQUIRE_EQUAL(r.ec, cluster::errc::success);
-        auto md = get_local_cache().get_topic_metadata(r.tp_ns);
+        auto md = app->metadata_cache.local().get_topic_metadata(r.tp_ns);
 
         BOOST_REQUIRE_EQUAL(md.has_value(), true);
-        wait_for_leaders(cntrl->get_partition_leaders().local(), *md);
+        wait_for_leaders(app->controller->get_partition_leaders().local(), *md);
         BOOST_REQUIRE_EQUAL(md.value().tp_ns, r.tp_ns);
     }
 }
 
 FIXTURE_TEST(test_autocreate_on_non_leader, cluster_test_fixture) {
     // root cluster node
-    model::node_id n_1(1);
-    model::node_id n_2(2);
-
-    add_controller(n_1, ss::smp::count, 9092, 11000, {});
-    add_controller(
-      n_2,
-      ss::smp::count,
-      9093,
-      11001,
-      {{.addr = unresolved_address("127.0.0.1", 11000)}});
+    model::node_id n_1(0);
+    model::node_id n_2(1);
```

Comment:
why did the node_id's change

Reply:
there is automated handling for first node with `id == 0` to be a cluster root.